src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/TaskListTabPanel.tsx

```diff
@@ -221,11 +231,45 @@ const TaskListTabPanel = (props: TaskListTabPanelProps) => {
       deadline: datePipe(task.deadline),
       priority: task.priority,
       assignee: assigneeString.substring(0, assigneeString.length - 2),
+      task: task,
       taskId: task.taskId
     };
   });
 
-  if (!auth.user) return <LoadingIndicator />;
+  const { isLoading, isError, mutateAsync, error } = useEditTask();
+  const {
+    isLoading: assigneeIsLoading,
+    isError: assigneeIsError,
+    mutateAsync: assigneeMutateAsync,
+    error: assigneeError
+  } = useEditTaskAssignees();
+
+  if (isLoading || assigneeIsLoading || !auth.user) return <LoadingIndicator />;
+  if (isError) return <ErrorPage message={error?.message} />;
+  if (assigneeIsError) return <ErrorPage message={assigneeError?.message} />;
+
+  const handleEditTask = async ({ taskId, notes, title, deadline, assignees, priority }: FormInput) => {
+    if (auth.user?.userId === undefined) throw new Error('Cannot edit a task while not being logged in');
+    await mutateAsync({
+      taskId,
+      notes,
+      title,
+      deadline,
+      priority
+    }).catch((error) => {
+      toast.error(error.message);
+      throw new Error(error);
```

Comment:
the whole point of toasting the error is so that we dont throw it and crash the page